api/server.go

```diff
@@ -59,10 +71,15 @@ func (c apiServer) Start(ctx context.Context, wg *sync.WaitGroup, startupErr cha
 		InsecureSkipVerify: true,
 	})
 
-	opts := []grpc.ServerOption{grpc.Creds(creds)}
+	// set http error interceptor
+	runtime.HTTPError = httpResponseInterceptor
 
-	grpcServer := grpc.NewServer(opts...)
+	opts := []grpc.ServerOption{
+		grpc.Creds(creds),
+		// grpcInterceptor(), // enable this once we start requiring the tenant id to passed as header
```

Comment:
I would add it rather than have a many commented lines of code in the code base.
These are called feature flags. you can read about it at https://martinfowler.com/articles/feature-toggles.html

Reply:
Feature flags are used to enable features, not for integrations which is what i'm talking about here. Feature flags which enable/disable a feature should not break the application flow, whereas this one does. I would rather keep it commented rather than putting it behind a flag which enables makes the application error out for every API call. Hence, I would like to keep this commented as is and enable it when our API require tenantID to be passed.

We have used feature flags earlier like `pprof` but this one doesn't belong under feature flag IMO.

Reply:
Let's take it offline if you have any more concerns 👍